# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/azahar/hotfix-gcc.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

From a7bb469c7a695cebd7d69befd52339cd61a1daa2 Mon Sep 17 00:00:00 2001
From: PabloMK7 <hackyglitch2@gmail.com>
Date: Thu, 22 Jan 2026 12:47:22 +0100
Subject: [PATCH] log: Fix compilation with gcc backtrace

---
 src/common/logging/backend.cpp | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/src/common/logging/backend.cpp b/src/common/logging/backend.cpp
index 71dc1564f..36fcfeccf 100644
--- a/src/common/logging/backend.cpp
+++ b/src/common/logging/backend.cpp
@@ -330,9 +330,9 @@ class Impl {
             }
             backend_thread.request_stop();
             backend_thread.join();
-            const auto signal_entry =
-                CreateEntry(Class::Log, Level::Critical, "?", 0, "?",
-                            fmt::vformat("Received signal {}", fmt::make_format_args(sig)));
+            const auto signal_entry = CreateEntry(
+                Class::Log, Level::Critical, "?", 0, "?",
+                fmt::vformat("Received signal {}", fmt::make_format_args(sig)), time_origin);
             ForEachBackend([&signal_entry](Backend& backend) {
                 backend.EnableForStacktrace();
                 backend.Write(signal_entry);
@@ -345,12 +345,13 @@ class Impl {
                     abort();
                 }
                 line.pop_back(); // Remove newline
-                const auto frame_entry =
-                    CreateEntry(Class::Log, Level::Critical, "?", 0, "?", std::move(line));
+                const auto frame_entry = CreateEntry(Class::Log, Level::Critical, "?", 0, "?",
+                                                     std::move(line), time_origin);
                 ForEachBackend([&frame_entry](Backend& backend) { backend.Write(frame_entry); });
             }
             using namespace std::literals;
-            const auto rip_entry = CreateEntry(Class::Log, Level::Critical, "?", 0, "?", "RIP"s);
+            const auto rip_entry =
+                CreateEntry(Class::Log, Level::Critical, "?", 0, "?", "RIP"s, time_origin);
             ForEachBackend([&rip_entry](Backend& backend) {
                 backend.Write(rip_entry);
                 backend.Flush();

